src/rdkafka_offset.c

```diff
@@ -791,8 +791,13 @@ void rd_kafka_offset_reset (rd_kafka_toppar_t *rktp, int64_t err_offset,
                 return;
         }
 
-	if (err_offset == RD_KAFKA_OFFSET_INVALID || err)
+	if (err_offset == RD_KAFKA_OFFSET_INVALID || err) {
 		offset = rktp->rktp_rkt->rkt_conf.auto_offset_reset;
+                if (rktp->rktp_rkt->rkt_rk->rk_conf.isolation_level ==
+                    RD_KAFKA_READ_COMMITTED && 
+                    offset == RD_KAFKA_OFFSET_END)
+                        offset = RD_KAFKA_OFFSET_LAST_STABLE;
+        }
```

Comment:
`} else` on the same line

Reply:
as noted earlier, i've removed this logic. you should definitely double check that.